### PR TITLE
User & FoodSpot Validation and Exception Handling

### PR DESCRIPTION
**Pull request by sahedw**
- create validations in classes and mark controller endpoints with validation
- exception handling in backend with existing username and invalid requests
- add validation integration tests for controller
- add toasts for ux hence better feedback for user
- toast for not met requirements only general
- create validation for signup page with toast. Toast also in login
- finish validation from login page and baxkbutton from signup page
- fix
- add validation to foodspot add form and exception handling

### CODE REVIEW

**Review by StormTrooper-007 — APPROVED** [review]

File: backend/src/main/java/com/github/sahedw/backend/controllers/FoodSpotUserController.java [hunk 1]
```diff
@@ -25,16 +33,28 @@ public String getMe() {
...
 
     @PostMapping("/sign-up")
-    public String signUp(@RequestBody FoodSpotUserForSignUp dtoUser){
+    public String signUp(@Valid @RequestBody FoodSpotUserForSignUp dtoUser) {
         return foodSpotUserService.signUp(dtoUser);
     }
+
+    @ExceptionHandler({UsernameAlreadyExistsException.class})
+    @ResponseStatus(HttpStatus.CONFLICT)
+    public ErrorMessage handleUsernameAlreadyExistsExceptions(UsernameAlreadyExistsException exception) {
```
Comment (Satttoshi) [inline backend/src/main/java/com/github/sahedw/backend/controllers/FoodSpotUserController.java:50]: nice

File: backend/src/main/java/com/github/sahedw/backend/models/FoodSpot.java [hunk 2]
```diff
@@ -14,8 +16,16 @@
...
+
+    @NotBlank
+    @Size(min=2, max=150)
     private String name;
+
+    @NotBlank
+    @Size(min=5, max=100)
     private String address;
+
+    @NotBlank
```
Comment (Satttoshi) [inline backend/src/main/java/com/github/sahedw/backend/models/FoodSpot.java:28]: seems valid

File: backend/src/test/java/com/github/sahedw/backend/controllers/FoodSpotControllerTest.java [hunk 3]
```diff
@@ -154,4 +155,48 @@ void expectDeletedFoodSpot_whenDeleteRequestIsCalled() throws Exception {
...
+                                    "name": "   ",
+                                    "address": "Steindamm 58",
+                                    "category": "DOENER"
+                                }
+                                """)
+                        .with(csrf()))
+                .andExpect(MockMvcResultMatchers.status().isUnprocessableEntity())
+                .andExpect(MockMvcResultMatchers.content().json("""
+                        {
+                            "message": "must not be blank"
```
Comment (Satttoshi) [inline backend/src/test/java/com/github/sahedw/backend/controllers/FoodSpotControllerTest.java:198]: Aceuuuuu 😭

File: frontend/src/App.tsx [hunk 4]
```diff
@@ -88,9 +148,34 @@ function App() {
...
+                        border: '2px solid #713200',
+                        padding: '10px',
+                        color: 'black',
+                        boxShadow: "8px 8px 0px -2px #000000",
+                        backgroundColor: "orangered"
+
+                    }
+                })
             });
     }
```
Comment (Satttoshi) [inline frontend/src/App.tsx:181]: Maybe Refactor fetching? This File looks brutal like Akeinu

File: frontend/src/components/SignUpPage.tsx [hunk 5]
```diff
@@ -45,56 +66,80 @@ function SignUpPage({onRegistration}: Props) {
...
+                                <li className={repeatedPassword !== password
+                                && repeatedPassword.trim().length === 0
+                                    ? "invalid" : "valid"}>
+                                    Must be identical
+                                </li>
+                            </ul>
+                            <SeePassword className={"password-icon"} size={"1.5em"}
+                                         onShowPassword={handleShowRepeatedPassword}
+                                         currentShowValue={showRepeatedPassword}/>
+                        </section>
```
Comment (Satttoshi) [inline frontend/src/components/SignUpPage.tsx:134]: übertreib

**Review by Satttoshi — COMMENTED** [review]
lgtm :rocket:

please refactor frontend components

File: backend/src/test/java/com/github/sahedw/backend/controllers/FoodSpotControllerTest.java (hunk 3 above)
Comment (sahedw) [reply backend/src/test/java/com/github/sahedw/backend/controllers/FoodSpotControllerTest.java:198]: 😭

File: frontend/src/components/SignUpPage.tsx (hunk 5 above)
Comment (sahedw) [reply frontend/src/components/SignUpPage.tsx:134]: 😪

**Pull request merged**